src/main/java/com/semmle/jira/addon/LgtmServlet.java

```
@@ -43,94 +45,119 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
-      sendError(resp, HttpServletResponse.SC_GONE, "Issue " + request.issueId + " not found.");
-      return;
-    }
+    if (request.transition == Request.Transition.CREATE) {
+      try {
+        Long issueId = createIssue(request, config);
+        Response response = new Response(issueId);
+        sendJSON(resp, HttpServletResponse.SC_CREATED, response);
+      } catch (CustomFieldRetrievalException e) {
+        log.error("Retrieval of custom field for config key failed.", e);
+        sendError(
+            resp,
+            HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+            "Retrieval of configuration key failed. Please check your add-on configuration.");
+      } catch (CreateIssueException e) {
+        sendError(resp, HttpServletResponse.SC_INTERNAL_SERVER_ERROR, e.getMessage());
+      }
+    } else {
+      MutableIssue issue =
+          ComponentAccessor.getIssueService().getIssue(user, request.issueId).getIssue();
+      if (issue == null) {
+        sendError(resp, HttpServletResponse.SC_GONE, "Issue " + request.issueId + " not found.");
+        return;
+      }
 
-    switch (request.transition) {
-      case CREATE:
-        createIssue(jsonRequest, request, resp, config);
-        break;
-      case REOPEN:
-        applyTransition(issue, Constants.WORKFLOW_REOPEN_TRANSITION_NAME, true, user, resp);
-        break;
-      case CLOSE:
-        applyTransition(issue, Constants.WORKFLOW_CLOSE_TRANSITION_NAME, true, user, resp);
-        break;
-      case SUPPRESS:
-        applyTransition(issue, Constants.WORKFLOW_SUPPRESS_TRANSITION_NAME, false, user, resp);
-        break;
-      case UNSUPPRESS:
-        applyTransition(issue, Constants.WORKFLOW_REOPEN_TRANSITION_NAME, true, user, resp);
-        break;
+      boolean skipValidation = request.transition != Transition.SUPPRESS;
+      String transitionName = null;
+      switch (request.transition) {
+        case CREATE:
+          // Already handled
```

This comment is a bit funny. If we end-up in this case that means it actually wasn't handled. Perhaps better to throw an exception or just leave this case out, as the default case handles it quite well.